presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestReorderJoins.java

```diff
@@ -52,12 +61,139 @@ public void testChoosesAJoin()
                 .matches(join(
                         JoinNode.Type.INNER,
                         ImmutableList.of(equiJoinClause("B1", "C1"), equiJoinClause("A1", "C1")),
-                        join(
+                        Optional.empty(),
+                        Optional.of(PARTITIONED), join(
                                 JoinNode.Type.INNER,
                                 ImmutableList.of(equiJoinClause("A1", "B1")),
-                                values(ImmutableMap.of("A1", 0)),
-                                values(ImmutableMap.of("B1", 0)
-                                )),
-                        values(ImmutableMap.of("C1", 0))));
+                                Optional.empty(),
+                                Optional.of(PARTITIONED), values(ImmutableMap.of("A1", 0)),
+                                values(ImmutableMap.of("B1", 0))
+                        ),
+                        values(ImmutableMap.of("C1", 0))
+                ));
+    }
+
+    @Test
+    public void testReplicatesAndFlipsWhenOneTableMuchSmaller()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("tiny")
+                .setSystemProperty("join_distribution_type", "automatic")
+                .build();
+        LocalQueryRunner queryRunner = queryRunnerWithDummyNodeCount(session, 4);
+        new RuleTester(queryRunner)
+                .assertThat(new ReorderJoins(new CostComparator(1, 1, 1)))
+                .on(p ->
+                        p.joinGraph(
+                                ImmutableList.of(
+                                        p.values(new PlanNodeId("values1"), p.symbol("A1", BIGINT)),
```

Comment:
I suggest naming the values after the symbols they produce. So for example the Values node 'valuesA' produces symbols 'A1, A2, A3, ...'. I think that would make these unit tests and the other ones in this PR more readable.